tests/testthat/test-plot_seromodel.R

```diff
@@ -0,0 +1,313 @@
+          y = "seroprev", size = "n_sample"
+        ), geom_params = list(na.rm = FALSE)
+      )
+    ),
+    coordinates = list(limits = list(x = c(1, 60), y = c(
+      0.3,
+      1
+    ))),
+    labels = list(
+      x = "Age", y = "Seroprevalence", ymin = "seroprev_lower",
+      ymax = "seroprev_upper", size = "n_sample"
+    )
+  )
+
+  expect_lists_equal_with_tolerance(expected_plot, actual_plot)
+})
+
+
+
+test_that("plot_serosurvey creates a binned ggplot with correct structure", {
+  skip_on_cran()
+
+  seromodel <- fit_seromodel(
+    serosurvey = serosurvey
+  )
+  actual_plot <- extract_plot_data(plot_serosurvey(serosurvey, bin_serosurvey = TRUE, bin_step = 10))
+
+  prepared_serosurvey <- create_prepared_serosurvey(serosurvey)
+
+  expected_plot <- list(
+    classes = c("gg", "ggplot"),
+    layers = list(
+      list(
+        mapping = list(
+          ymin = "seroprev_lower", ymax = "seroprev_upper"
+        ),
+        geom_params = list(
+          na.rm = FALSE, orientation = NA, width = 0.1
+        )
+      ),
+      list(
+        mapping = list(
+          y = "seroprev", size = "n_sample"
+        ),
+        geom_params = list(na.rm = FALSE)
+      )
+    ),
+    coordinates = list(
+      limits = list(x = c(2, 60), y = c(
+        0.3,
+        1
+      ))
+    ),
+    labels = list(
+      x = "Age", y = "Seroprevalence", ymin = "seroprev_lower",
+      ymax = "seroprev_upper", size = "n_sample"
+    )
+  )
+
+  expect_lists_equal_with_tolerance(expected_plot, actual_plot)
+})
+
+test_that("plot_summary creates a ggplot with correct structure", {
+  skip_on_cran()
+
+  seromodel <- fit_seromodel(
+    serosurvey = serosurvey
+  )
+  loo_estimate_digits <- 1
+  central_estimate_digits <- 2
+  seroreversion_digits <- 2
+  rhat_digits <- 2
+  size_text <- 11
+
+  plot <- plot_summary(
+    seromodel = seromodel,
+    serosurvey = serosurvey,
+    loo_estimate_digits = loo_estimate_digits,
+    central_estimate_digits = central_estimate_digits,
+    rhat_digits = rhat_digits,
+    size_text = size_text
+  )
+  actual_data <- plot$data
+
+  actual_plot <- extract_plot_data(plot)
+
+  expected_data <- data.frame(
+    row = c(5, 4, 3, 2, 1),
+    text = c(
+      "model_name: constant_no_seroreversion",
+      "elpd_loo: -23.9(se=12.2)",
+      "foi: 0.082(95% CI, 0.063-0.1)",
+      "foi_rhat: 1",
+      "converged: yes"
+    )
+  )
+
+  expected_plot <- list(
+    classes = c("gg", "ggplot"),
+    layers = list(list(
+      mapping = list(
+        label = "text"
+      ),
+      geom_params = list(
+        parse = FALSE, check_overlap = FALSE,
+        size.unit = "mm", na.rm = FALSE
+      )
+    )),
+    coordinates = list(
+      limits = list(
+        x = NULL, y = NULL
+      )
+    ),
+    labels = list(x = "x", y = "row", label = "text")
+  )
+
+  # Checks that the actual plot data are at most 20% disimilar to the expected data
+  # using Levenshtein distance (adist)
+  expect_true({
```

Comment:
You may be able to avoid this using the approach above.

Reply:
Same as above, I think it's better to keep the values hardcoded, otherwise the test code might get too complex and hard to maintain if we make changes to the tested code